src/test/mocks/BeaconChainMock.t.sol

```diff
@@ -7,9 +7,10 @@ import "src/contracts/libraries/BeaconChainProofs.sol";
 import "src/contracts/libraries/Merkle.sol";
```

Comment:
Request - let's please move all the beacon chain / EIP mock contracts to a subfolder, for organization's sake.

e.g:

```
mocks/beacon/
 - BeaconChainMock.t.sol
 - BeaconChainMock_Deneb.t.sol
 - EIP_4788_Oracle_Mock.t.sol
```


I'm adding 2 more mocks for Pectra predeploys in MOOCOW, and having all of them in the same place is super helpful.

Reply:
oh yeah, and i also did some refactoring of my own and added two libraries:

![image](https://github.com/user-attachments/assets/8f2e0b03-dd17-410f-922f-067c6f8dab18)

The libraries also need to sit next to the beacon chain stuff. So, a subfolder is needed :D